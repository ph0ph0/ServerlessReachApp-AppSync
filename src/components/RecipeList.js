import React from 'react'
import { css } from 'glamor'
import Recipe from './Recipe'

const RecipeList = ({ data }) => {
    return (
        <div data-testid = "container" {...css(styles.container)}>    
            <h1>Recipes</h1>
            {
                data.listRecipes.items.map((recipe, index) => (
                    <Recipe 
                    recipe = {recipe}
                    index = {index}
                    />
                ))
            }
        </div>
    )
}

const styles = {
    title: {
        fontSize: 16
    },
    subtitle: {
        fontSize: 14,
        color: 'rgba(0, 0, 0, 0.5)'
    },
    recipe: {
        boxShadow: '2px 2px 5px rgba(0, 0, 0, 0.2)',
        marginBottom: 7,
        padding: 14,
        border: '1px solid #EDEDED'
    },
    container: {
        display: 'flex',
        flexDirection: 'column',
        paddingLeft: 100,
        paddingRight: 100,
        textAlign: 'left',
        alignItems: 'center'
    }
}

export default RecipeList