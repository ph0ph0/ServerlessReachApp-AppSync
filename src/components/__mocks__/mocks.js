
import ListRecipes from '../../graphql/queries/ListRecipes'

const mocks = [
    {
        request: {
            query: ListRecipes
        }, 
        result: {
            data: {
                listRecipes: {
                    items: [
                        {
                            name: "Tuna Rolll",
                            id: "1",
                            instructions: ["Mix Tuna"],
                            ingredients: ["Tuna"]
                        }  
                    ]                  
                }
            }
        }
    }
]

const data = {
    listRecipes: {
        items: [
            {
                id: "1",
                name: "Tuna Roll",
                instructions: ["Mix Tuna"],
                ingredients: ["Tuna"]
            }  
        ]                  
    }
}


export default mocks
export {data}