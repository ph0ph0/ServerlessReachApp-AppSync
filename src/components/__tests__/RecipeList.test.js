import React from 'react'

import RecipeList from '../RecipeList'

import {render, cleanup, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { MockedProvider } from '@apollo/react-testing'
import mocks, {data}  from '../__mocks__/mocks'

afterEach(cleanup)

it("renders Recipe", async () => {
    const { getByTestId } = render(
        <MockedProvider mocks = {mocks} addTypename = {false}>
            <RecipeList data = {data}/>
        </MockedProvider>
    )

    const recipeContainer = await waitForElement(() => getByTestId("container"))

    expect(recipeContainer).toBeInTheDocument()

    expect(getByTestId("recipeName")).toHaveTextContent("Recipe name: Tuna Roll")
})

it("Recipe List renderss", () => {
    const { asFragment } = render(
        <MockedProvider mocks = {mocks} addTypename = {false}>
                <RecipeList data = {data}/>
        </MockedProvider>
    )
    expect(asFragment()).toMatchSnapshot()
})