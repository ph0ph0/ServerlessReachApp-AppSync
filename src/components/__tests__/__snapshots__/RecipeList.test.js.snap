// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe List renderss 1`] = `
<DocumentFragment>
  <div
    data-css-l6r9g8=""
    data-testid="container"
  >
    <h1>
      Recipes
    </h1>
    <div
      data-css-3nuvy1=""
    >
      <p
        data-css-1sfx3dj=""
        data-testid="recipeName"
      >
        Recipe name: Tuna Roll
      </p>
      <div>
        <p
          data-css-1sfx3dj=""
        >
          Ingredients
        </p>
        <p
          data-css-ar9c07=""
          data-testid="recipeIngredients"
        >
          Tuna
        </p>
      </div>
      <div>
        <p
          data-css-1sfx3dj=""
        >
          Instructions
        </p>
        <p
          data-css-ar9c07=""
          data-testid="recipeInstructions"
        >
          1. Mix Tuna
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
